components/kyma-environment-broker/internal/storage/postsql/read.go

```diff
@@ -429,6 +429,35 @@ func (r readSession) GetLatestRuntimeStateWithReconcilerInputByRuntimeID(runtime
 	return state, nil
 }
 
+func (r readSession) GetLatestRuntimeStatesByRuntimeID(runtimeID string, n int) ([]dbmodel.RuntimeStateDTO, dberr.Error) {
+	var states []dbmodel.RuntimeStateDTO
+	condition := dbr.And(dbr.Eq("runtime_id", runtimeID), dbr.Or(
+		dbr.And(dbr.Neq("cluster_setup", nil), dbr.Neq("cluster_setup", "")),
+		dbr.And(dbr.Neq("kyma_config", nil), dbr.Neq("kyma_config", "")),
+	))
+
+	count, err := r.session.
+		Select("*").
+		From(RuntimeStateTableName).
+		Where(condition).
+		OrderDesc(CreatedAtField).
+		// because both cluster_setup as well as kyma_config are encrypted
+		// we can't easily query the content of the json. This gets the last
+		// N operations for further processing
```

Comment:
I was trying to use https://www.postgresql.org/docs/8.3/pgcrypto.html instead and query on decrypted json object but I couldn't get it to work the same way the encryption works in KEB go code. The reason for this is that for each operation, we store some `kyma_config`. Frequently, after decoding, it is just `{ <nil> [] [] <nil>}` but it makes the query harder to implement because the content is encrypted data.

Reply:
TODO: we will populate `runtime_state.kymaVersion` in the database retrospectively for all relevant entries and use that in the future.